File: iis_web/iis_web/Models/Character.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Xml.Serialization;

namespace iis_web.Models
{
    [XmlRoot(ElementName = "character")]
    public class Character
    {
        public enum CharacterStatus
        {
            ALIVE,
            DEAD,
            UNKNOWN
        }

        public enum CharacterGender
        {
            UNKNOWN,
            FEMALE,
            MALE,
            GENDERLESS
        }

        [XmlIgnore]
        [XmlAttribute(AttributeName = "id")]
        public int ID { get; set; }

        [XmlElement(ElementName = "name")]
        public string Name { get; set; }

        [XmlElement(ElementName = "status")]
        public string Status { get; set; }

        [XmlElement(ElementName = "type")]
        public string Type { get; set; }

        [XmlElement(ElementName = "species")]
        public string Species { get; set; }

        [XmlElement(ElementName = "gender")]
        public string Gender { get; set; }

        [XmlElement(ElementName = "origin")]
        public Location Origin { get; set; }

        [XmlElement(ElementName = "location")]
        public Location Location { get; set; }

        [XmlElement(ElementName = "image")]
        public string Image { get; set; }

        [XmlElement(ElementName = "episodes")]
        public Episodes Episodes { get; set; }
    }
}

File: main_iis/src/main/java/hr/algebra/iis/model/Character.java
package hr.algebra.iis.model;


import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;

import javax.xml.bind.annotation.*;
import java.io.Serializable;
import java.util.List;
import java.util.Set;

@XmlType(namespace = "http://www.w3.org/2001/character")
@XmlRootElement(name = "character")
@Entity
@Table(name = "character")
public class Character implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    int id ;

    String name;

    @Enumerated(EnumType.ORDINAL)
    Status status;

    String species;

    String type;

    @Enumerated(EnumType.ORDINAL)
    Gender gender;

    @OneToOne(cascade = CascadeType.MERGE)
    Location origin;

    @ManyToOne(cascade = CascadeType.MERGE)
    @JoinColumn(name = "location_id", nullable = false)
    Location location;

    String image;

    @ManyToMany(cascade = CascadeType.MERGE)
    @JoinTable(
            name = "character_episode",
            joinColumns = @JoinColumn(name = "character_id"),
            inverseJoinColumns = @JoinColumn(name = "episode_id")
    )
    List<Episode> episodes;

    @XmlAttribute
    public int getId() {
        return id;
    }

    @XmlElement(name = "name")
    public String getName() {
        return name;
    }

    @XmlElement(name = "status")
    public Status getStatus() {
        return status;
    }

    @XmlElement(name = "species")
    public String getSpecies() {
        return species;
    }

    @XmlElement(name = "type")
    public String getType() {
        return type;
    }

    @XmlElement(name = "gender")
    public Gender getGender() {
        return gender;
    }

    @XmlElement(name = "origin")
    @JsonManagedReference
    public Location getOrigin() {
        return origin;
    }

    @XmlElement(name = "location")
    @JsonManagedReference
    public Location getLocation() {
        return location;
    }

    @XmlElement(name = "image")
    public String getImage() {
        return image;
    }

    @XmlElementWrapper(name = "episodes")
    @XmlElement(name = "episode")
    @JsonManagedReference
    public List<Episode> getEpisodes() {
        return episodes;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setStatus(Status status) {
        this.status = status;
    }

    public void setSpecies(String species) {
        this.species = species;
    }

    public void setType(String type) {
        this.type = type;
    }

    public void setGender(Gender gender) {
        this.gender = gender;
    }

    public void setOrigin(Location origin) {
        this.origin = origin;
    }

    public void setLocation(Location location) {
        this.location = location;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public void setEpisodes(List<Episode> episodes) {
        this.episodes = episodes;
    }

    public enum Gender {

        UNKNOWN,

        FEMALE,

        MALE,

        GENDERLESS
    }

    public enum Status {

        UNKNOWN,

        ALIVE,

        DEAD
    }
}
